addon/components/file-picker.js

```diff
@@ -51,6 +51,12 @@ export default Component.extend({
     this.$('.file-picker__input').on(
       'change', bind(this, 'filesSelected')
     );
+
+    // reset the input's value to trigger the `onchange` event in the case that the
+    // same file path is selected.
+    document.querySelector('.file-picker__input').addEventListener('click', (e) => {
+      e.target.value = '';
+    });
```

Comment:
Omitting jQuery assuming this addon is going to head in that direction.

Reply:
Does a click trigger the `onchange` or does the `onchange` only fire on user input?